spec/forms/signin/validate_mobile_token_form_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+RSpec.describe Signin::ValidateMobileTokenForm, type: :model do
+  subject(:form) { described_class.new(attrs) }
+  let(:attrs) { { user: user, code: signin_code.code } }
+  let(:user) { create(:user) }
+  let(:signin_code) { create(:signin_code, user:) }
+
+  context 'validations' do
+    it { should validate_presence_of(:user) }
+
+    it_behaves_like 'form with sign in code validations'
+  end
+
+  describe '#submit' do
+    subject(:submit) { form.submit }
+
+    context 'when attributes are valid' do
+      before do
+        allow(Signin::GenerateSigninCode).to receive(:call)
+          .and_return(generated_code)
+      end
+      let(:generated_code) { build(:signin_code).code }
+
+      it 'updates signin code used timestamp' do
+        expect { submit }.to change { signin_code.reload.used_at }
+      end
+
+      it 'generates a new signin code with fast expiration' do
+        expect(Signin::GenerateSigninCode).to receive(:call).with(
+          user:,
+          expiration_time: described_class::EXPIRATION_TIME_IN_SECONDS
+        )
+
+        submit
+      end
+
+      it 'returns the new generated code' do
+        expect(submit).to eq(generated_code)
+      end
+
+      context 'and the code generation raises an error' do
+        before do
+          allow(Signin::GenerateSigninCode).to receive(:call)
+            .and_raise(ActiveRecord::ConnectionTimeoutError)
+        end
+
+        it 'leaves the sigin code unused' do
```

Comment:
Typo
```suggestion
        it 'leaves the signin code unused' do
```

Reply:
You have NO IDEA how many times I miswrote it. :joy: 

faf513618ebe1433b648d4582aaf587a659762ba